lib/Net/Stripe/TypeConstraints.pm

```diff
@@ -0,0 +1,32 @@
+package Net::Stripe::TypeConstraints;
+
+use strict;
+use Moose::Util::TypeConstraints qw(subtype as where message);
+use DateTime qw();
+
+subtype 'Net::Stripe::Types::api_version',
+    as 'Str',
+    where {
+        my $min = '2012-10-26';
+        my $max = DateTime->now->ymd('-');
+        return if $_ !~ /^\d{4}-\d{2}-\d{2}$/;
+        return if $_ lt $min;
```

Comment:
I don't feel completely comfortable with that kind of date comparison. I think it would be preferable to get a `DateTime` object from `$_` and then use `<`, `>` comparison. See documentation around

> This module explicitly overloads the addition (+), subtraction (-), string and numeric comparison operators.

at https://metacpan.org/pod/DateTime

Reply:
agreed. will address shortly.

Reply:
updated